packages/babel-plugin-transform-parameters/test/fixtures/parameters/arrow-rest-super-no-classes-plugin/input.js

```diff
@@ -0,0 +1,6 @@
+class Bar extends Foo {
+    constructor() {
+        let f = (...args) => super(...args);
```

Comment:
Note that the input here is very similar to how we would transform `super()`:

https://github.com/babel/babel/blob/9862d4273d12383b14934a1b1ec8424f1d3a00f9/packages/babel-traverse/test/arrow-transform.js#L53-L71

My gut is that it should have been fixed in how we generate the super call:

https://github.com/babel/babel/blob/9862d4273d12383b14934a1b1ec8424f1d3a00f9/packages/babel-traverse/src/path/conversion.ts#L652

We already have an arrow function check when constructing `supercall`, 

https://github.com/babel/babel/blob/9862d4273d12383b14934a1b1ec8424f1d3a00f9/packages/babel-traverse/src/path/conversion.ts#L287-L291

somehow it does not throw, probably because `allowInsertArrow` is not properly passed.

Reply:
Perhaps we can add `allowInsertArrow: true` here?

https://github.com/babel/babel/blob/073b3b0536812bac7bb9d5d6363adef1dd174d07/packages/babel-plugin-transform-parameters/src/index.ts#L48-L49

Reply:
I tried to add `allowInsertArrow: false` and it fails on #13939 related tests...:thinking:
Maybe we need something like `allowInsertArrowWithRest`?

```
    × 13939
    × 13939 complex
    × 13939 private
    × 13939 private complex

SyntaxError: babel/packages/babel-plugin-transform-parameters/test/fixtures/regression/13939/input.js: Unable to transform arrow inside class property
      1 | class A {
    > 2 |   handle = (x = 0) => {
        |            ^
      3 |     console.log(x);
      4 |   };
      5 | }

...
```